### DISCUSSION TITLE
Using latest versions of Kotlin and Compose

### DISCUSSION THREAD

**Discussion by malliaridis** [arkivanov/Decompose#231] [Q&A] [answered]
I was wondering recently what one can expect for using the latest versions of Kotlin or Compose from those that are mentioned in the releases of Decompose. I was always updating to the latest stable versions, but I am not sure if that is a good practice or if that could cause some confusing errors in the future.

Like now, Kotlin 1.7.20 was released and I would like to update to work with the newest stable version, but if that is incompatible with Decompose 1.0.0-alpha-06 I should not update yet. The library is most of the time up-to-date and does a great job staying in sync with the latest releases, so there would be not really a drawback to wait a few days longer for the release supporting the newer versions.

**Comment by arkivanov** [comment, marked as answer]
You can update to the latest versions of Kotlin and Compose as long as the Compose version is binary compatible with the one used by Decompose. That's why I always specify used versions in release notes. Decompose doesn't use any of the experimental Compose API. Given the fact that Compose for Android and Desktop both maintain binary compatibly, everything should be safe. This is may not be the case with Compose for Web or iOS. Regardless, it's always recommended to check Compose release notes of versions between the one used by Decompose and the one used in your projects.
